mungegithub/github/github.go

```diff
@@ -630,14 +658,26 @@ func (config *Config) needsBranchProtection(branch *github.Branch, contexts []st
 
 // SetBranchProtection protects a branch and sets the required contexts
 func (config *Config) SetBranchProtection(name string, contexts []string) error {
-	branch, resp, err := config.client.Repositories.GetBranch(config.Org, config.Project, name)
-	config.analytics.GetBranch.Call(config, resp)
+	prot, resp, err := config.client.Repositories.GetBranchProtection(
+		context.Background(),
+		config.Org,
+		config.Project,
+		name,
+	)
+	config.analytics.GetBranchProtection.Call(config, resp)
 	if err != nil {
-		glog.Errorf("Got error getting branch %s: %v", name, err)
-		return err
+		b, errRead := ioutil.ReadAll(resp.Body)
```

Comment:
Unfortunately both cases return the same status code and the response body has to be inspected. A cleaner way to do this would be to use GetBranch first to check if branch protection is enabled as indicated by the Branch.Protected field and only call GetBranchProtection if it is enabled. That would cost an extra API token though, is that worth it?